server.js

```diff
@@ -17,115 +26,119 @@ const options = {
 
+//TEST DATA
+const newTestNote = {
+  "title": "Yamaha SLB-200LTD Silent Bass",
+  "priority": 3,
+  "body": "The most popular Electric Upright Bass in the line-up."
+};
+
+
+//USING AXIOS REQUESTS 3-21-19
+Axios.get(mLabNotes)
+// .then( (response ) => { console.log(response)})
+.then( (response ) => { console.log(response.status)})
+.catch( (err) => { console.log(err)})
+
+Axios.post(mLabNotes, {newTestNote}) 
+Axios.post('https://www.gggggggle.com/search?q=trees', {
+  "title": "Yamaha SLB-200LTD Silent Bass",
+  "priority": 3,
+  "body": "The most popular Electric Upright Bass in the line-up."
+})
+.then((response) =>{}) 
+
+// Axios.put('https://www.gggggggle.com/search?q=trees', {
+//   name:'David',
+//   city: 'SFO'
+// })
+// .then((response) =>{})
+
+// Axios.delete('https://www.gggggggle.com/search?q=trees', {
+//   name:'David'})
+//   .then((response) =>{})
+
+  //NOTES FROM LECTURE
+  // Should always must return something.  Not required but if no return what's the point?
+    // asyncFunction()
+    // .then(anotherAsyncFunction) (firstThing) => {return someting}
+    // .then(OneMoreAsyncFunc) (something) => {return somethingElse}
+    // .catch(errorHandlingFunction)
+
+  // callback hell is a real thing.  Promises solves this.
+//===========================
+
+//ORIGINAL SERVER REQUESTS before 3-21-19
 server.use(express.json()) // bodyParser function for json payloads
 
 server.use(helmet())
-server.use(cors()); // ie between netlify, heroku and mlab
 
-// const memCache ={}; 
+// Allow Cross-origin Resource Sharing i.e. between netlify, heroku and mlab
+// server.use(cors());
+server.use((req, res, next) =>{
+  res.header("Access-Control-Allow-Origin", "*");  //`*` allows all sites to make requests.  change to specific domains to restrict access.
+  res.header("Access-Control-Allow-Headers", "Origin, X-Requested-With, Content-Type, Accept, Authorization");
+  if (req.method === 'OPTIONS') {
+    res.header('Access-Control-Allow-Methods', 'PUT, POST, PATCH,DELETE, GET')
+    return res.status(200).json({})
+  }
+  next();  // allows other routes to take over.
+})
 
 //add CRUD routes
 server.get('/', (req, res) => {
-  res.send('Hello from the express server'); // sanity check
+  res.send('Hello from the d2rd Notes back-end express server using MongoDB on mLab'); // sanity check
 });
 
-server.get('/ElectricUprights', (req, res) => {
-  ElectricUpright.find()
+server.get('/Notes', (req, res) => {
+  Note.find()
     .then((data) => {
       res.json(data)
     })
     .catch(err => console.log(err.message))
 })
 
-server.post('/ElectricUprights/create', (req, res) => {
-  const { title, priority, body, price, itemURL, reviewURL } = req.body;
-  const myNote = { title, priority, body, price, itemURL, reviewURL };
-  const newNote = new ElectricUpright(myNote)
+server.post('/Notes/create', (req, res) => {
+  const { title, body } = req.body;
+  const myNote = { title, body };
+  const newNote = new Note(myNote)
```

Comment:
I would go back and take a look at documentation for creating a post request. This looks like a LOT of vars to me in one request.

Reply:
Are you suggesting that I only need ONE var for this?  And, as written, its convoluted?